### PR TITLE
`Development`: Improve JPQL style for grading repositories

### PR DESCRIPTION
**Pull request by julian-christl**
#### General

- [X] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/#naming-conventions-for-github-pull-requests).
#### Server
- [X] I followed the [coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/server/).

### Motivation and Context

I just fixed the style in one repository so I decided to go through some more repos to make the codebase more consistent and clean.

### Description

I enforced capitalisation (SELECT instead of select), simple variable declaration (:var instead of :#{#var}, explicit enum values, consistent newlines and indentations

#### Code Review
- [x] Code Review 1
- [ ] Code Review 2

### Test Coverage

No functional changes.

## Summary by CodeRabbit

- **Refactor**
    - Simplified query parameter binding in the `BonusRepository` for improved performance.
    - Enhanced readability of repository queries in various interfaces without changing functionality.
    - Updated JPQL syntax in the `RatingRepository` to ensure accurate calculations of average ratings.
    - Improved query structure in the `ResultRepository` for better clarity and added conditions for exercise types.
    - Adjusted query indentation in the `StudentScoreRepository` for improved code alignment.

### CODE REVIEW

**Review by Strohgelaender — DISMISSED** [review]
server tests are currently failing.

File: src/main/java/de/tum/in/www1/artemis/repository/GradingCriterionRepository.java [hunk 1]
```diff
@@ -17,6 +17,11 @@ public interface GradingCriterionRepository extends JpaRepository<GradingCriteri
 
     List<GradingCriterion> findByExerciseId(long exerciseId);
 
-    @Query("select distinct criterion from GradingCriterion criterion left join fetch criterion.structuredGradingInstructions where criterion.exercise.id = :#{#exerciseId}")
+    @Query("""
+            SELECT DISTINCT criterion
+            FROM GradingCriterion criterion
+                LEFT JOIN FETCH criterion.structuredGradingInstructions
+            WHERE criterion.exercise.id = :exerciseId
+            """)
```
[the remark this answers is not in the document]
Comment (julian-christl) [reply src/main/java/de/tum/in/www1/artemis/repository/GradingCriterionRepository.java:25]: They all are uppercase

File: src/main/java/de/tum/in/www1/artemis/repository/ResultRepository.java [hunk 2]
```diff
@@ -58,16 +58,19 @@ public interface ResultRepository extends JpaRepository<Result, Long> {
...
                 LEFT JOIN FETCH r.feedbacks f
                 LEFT JOIN FETCH f.testCase
-            WHERE r.completionDate =
-                (SELECT max(rr.completionDate) FROM Result rr
-                    WHERE rr.assessmentType = 'AUTOMATIC'
-                        AND rr.participation.exercise.id = :exerciseId
-                        AND rr.participation.student.id = r.participation.student.id
+                LEFT JOIN TREAT (r.participation AS ProgrammingExerciseStudentParticipation) sp
+            WHERE r.completionDate = (
+                    SELECT MAX(rr.completionDate)
```
Comment (nityanandaz) [inline src/main/java/de/tum/in/www1/artemis/repository/ResultRepository.java:63]: Maybe `r` could be `result` and `rr` could be `latest`

**Review by nityanandaz — COMMENTED** [review]
1 suggestion :)

File: src/main/java/de/tum/in/www1/artemis/repository/ResultRepository.java (hunk 2 above)
Comment (julian-christl) [reply src/main/java/de/tum/in/www1/artemis/repository/ResultRepository.java:63]: latest wouldn't really make sense as it's not just one result but multiple. We just take the max of all completion date, not the result itself.

We can of course argue about/improve the var naming here but that's an overall issue and out of scope of these PRs.
I hope you can appreciate this. Thank you for your review though!

**Review by nityanandaz — DISMISSED** [review]
Code review: looks good

**Review by Strohgelaender — DISMISSED** [review]

**Review by MaximilianAnzinger — APPROVED** [review]

**Pull request merged**